flink-java/src/main/java/org/apache/flink/api/java/utils/RequiredParameter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Facility to manage required parameters in user defined functions.
+ */
+public class RequiredParameter {
+
+	private static final String HELP_TEXT_PARAM_DELIMITER = "\t";
+	private static final String HELP_TEXT_LINE_DELIMITER = "\n";
+
+	private HashMap<String, Option> data;
+
+	public RequiredParameter() {
+		this.data = new HashMap<>();
+	}
+
+	public void add(Option option) throws RequiredParameterException {
```

Comment:
`req.add("name").type(...).values(...)`
oh, somehow I managed to miss that one.

I will add a String based version, maybe we can get a third opinion on the removal? I'd be in favor of giving the user the option to directly add an `Option` as well.